tests/unit/auth/permissionChecks.js

```diff
@@ -534,3 +554,307 @@ describe('validatePolicyConditions', () => {
         });
     });
 });
+
+describe('validateQuotas (buckets)', () => {
+    beforeEach(() => {
+        ScubaClientInstance.enabled = true;
+        ScubaClientInstance._getLatestmetricsCallback = sinon.stub().resolves({});
+    });
+
+    afterEach(() => {
+        sinon.restore();
+    });
+
+    it('should return null if quota is <= 0 or scuba is disabled', done => {
+        validateQuotas(mockBucketNoQuota, {}, [], '', false, mockLog, err => {
+            assert.ifError(err);
+            assert.strictEqual(ScubaClientInstance._getLatestmetricsCallback.called, false);
+            done();
+        });
+    });
+
+    it('should return null if scuba is disabled', done => {
+        ScubaClientInstance.enabled = false;
+        validateQuotas(mockBucket, {}, [], '', false, mockLog, err => {
+            assert.ifError(err);
+        assert.strictEqual(ScubaClientInstance._getLatestmetricsCallback.called, false);
+            done();
+        });
+    });
+
+    it('should return null if metrics retrieval fails', done => {
```

Comment:
side note (not for this PR in any case) : bypassing quota in that case may allow for easily bypassing the quotas with DOS attack... e.g. if we try to hammer the ser-ver with many putObject requests, scuba will probably fail before cloudserver.... and the request may bypass quota :/

Reply:
It can be worth discussing this PR. We have two things:
- The periodic healthchecks, they are the one that determine the state of the quota checks.
- The error handling, when evaluating quotas. When there is no quota and scuba is enabled, it means count items never ran for the current bucket/account, and thus we need to ignore the quotas. but for the other cases.

Today we ignore the quotas if the healthchecks failed, or if there is an error when getting the metrics (without discriminating between really missing metrics and other issues).

If the bucket/quota exists, we could try to still return an error in this case, but I'm not sure how.

But anyway, the scuba service must be sized properly, so that it is not a bottleneck. Mongo can be, but scuba shouldn't...